barretenberg/cpp/src/barretenberg/dsl/acir_proofs/honk_contract.hpp

```diff
@@ -2051,7 +2050,8 @@ abstract contract BaseHonkVerifier is IVerifier {
                 mem.scalingFactorPos = mem.batchingChallenge * mem.posInvertedDenominator;
                 mem.scalingFactorNeg = mem.batchingChallenge * tp.shplonkNu * mem.negInvertedDenominator;
                 // [Aₗ] is multiplied by -v^{2l}/(z-r^{2^l}) - v^{2l+1} /(z+ r^{2^l})
-                scalars[NUMBER_OF_ENTITIES + 1 + i] = mem.scalingFactorNeg.neg() + mem.scalingFactorPos.neg();
+                scalars[NUMBER_UNSHIFTED + 1 + i] =
+                    mem.scalingFactorNeg.neg() + mem.scalingFactorPos.neg();
```

Comment:
Unclear how expensive the `neg` is here, but might be cheaper for you to cache it instead